src/market.ts

```diff
@@ -1209,3 +1211,89 @@ function createAccountAccumulation(id: Bytes): AccountAccumulationStore {
+}
+
+function createMarketAccountAccumulator(
+  marketAccount: MarketAccountStore,
+  toVersion: BigInt,
+  collateral: BigInt,
+  fees: BigInt
+): void
+{
+  // FromID holds the current value for the accumulator
+  const fromId = buildMarketAccountAccumulatorId(marketAccount, marketAccount.latestVersion)
+  const toId = buildMarketAccountAccumulatorId(marketAccount, toVersion)
+  const fromAccumulator = MarketAccountAccumulatorStore.load(fromId)
+  let entity = new MarketAccountAccumulatorStore(toId)
+
+  entity.marketAccount = marketAccount.id
+  entity.fromVersion = marketAccount.latestVersion
+  entity.toVersion = toVersion
+
+  entity.collateral = collateral
+  entity.fees = fees
+
+  entity.save()
+}
+
+function createMarketAccountAccumulation(
+  marketAccount: MarketAccountStore,
+  toVersion: BigInt,
+  collateral: BigInt,
+  fees: BigInt
+): void {
+  let toId = buildMarketAccountAccumulatorId(marketAccount, toVersion)
+  const toAccumulator = loadMarketAccountAccumulator(toId)
+
+  const buckets = ['hourly', 'daily', 'weekly', 'all']
+  for (let i = 0; i < buckets.length; i++) {
+    const bucketTimestamp = timestampToBucket(toAccumulator.fromVersion, buckets[i])
+    let entity = loadOrCreateMarketAccountAccumulation(marketAccount, buckets[i], bucketTimestamp)
+
+    entity.collateral = entity.collateral.plus(collateral)
+    entity.fees = entity.fees.plus(fees)
+    entity.save()
+  }
+}
+
+function accumulateFulfilledOrder(
+  marketAccount: MarketAccountStore,
+  oracleVersionTimestamp: BigInt,
+  isDeltaNeutral: bool,
+  maker: BigInt,
+  long: BigInt,
+  short: BigInt,
+  price: BigInt
+): void {
+  const buckets = ['hourly', 'daily', 'weekly', 'all']
+  for (let i = 0; i < buckets.length; i++) {
+    const bucketTimestamp = timestampToBucket(oracleVersionTimestamp, buckets[i])
+
+    // Accumulate at MarketAccount
+    const marketAccountAccumulation = loadOrCreateMarketAccountAccumulation(marketAccount, buckets[i], bucketTimestamp)
+    if (!isDeltaNeutral) {
+      marketAccountAccumulation.trades = marketAccountAccumulation.trades.plus(BigInt.fromU32(1))
+    }
+    // Record absolute position, which was aggregated in fulfillOrder
+    marketAccountAccumulation.maker = maker
+    marketAccountAccumulation.long = long
+    marketAccountAccumulation.short = short
+    marketAccountAccumulation.makerNotional = notional(maker, price)
+    marketAccountAccumulation.longNotional = notional(long, price)
+    marketAccountAccumulation.shortNotional = notional(short, price)
+
+    // Accumulate at Market
+    const marketAccumulation = loadOrCreateMarketAccumulation(marketAccount.market, buckets[i], bucketTimestamp)
+    marketAccumulation.trades = marketAccumulation.trades.plus(BigInt.fromU32(1))
+    // If this is the MarketAccount's first trade for the bucket, increment the number of traders
+    if (!isDeltaNeutral && marketAccountAccumulation.trades == BigInt.fromU32(1)) {
```

Comment:
Lets use `.equals(BigInt.fromU32(1))`, I'm not sure how loose equality works in assemblyscript

Reply:
Sure; done in c8a9eb17807a33a9da08be431289f380d2a4431e.